apps/system/test/unit/card_test.js

```diff
@@ -82,368 +76,173 @@ suite('system/Card', function() {
     });
 
-    test('subTitle when private browser splash', function() {
-      var app = makeApp({
-        name: 'shortname',
-        origin: 'app://search.gaiamobile.org',
-        url: 'app://search.gaiamobile.org/newtab.html'
-      });
-      this.sinon.stub(app, 'isBrowser').returns(true);
-      var appCard = new Card({
-        app: app,
-        manager: mockManager
-      });
-      appCard.render();
+  });
 
-      assert.equal(appCard.subTitle, '');
-      assert.isFalse(appCard.element.classList.contains('show-subtitle'));
-    });
+});
 
-    test('gets icon from appWindow', function() {
-      var card = this.card;
-      var app = card.app;
-      card.render();
-      getIconPromise.mFulfillToValue({originalUrl: iconDataURI, blob: {}});
+suite('system/SwipeToKillMotion', function() {
+  var motion, el, setTranslateY;
+  var currentY;
 
-      assert.ok(app.getSiteIconUrl.called);
-      assert.ok(Icon.prototype.renderBlob.called);
-      assert.isTrue(card.iconValue.indexOf(iconDataURI) > -1);
+  setup(function() {
+    el = document.createElement('div');
+    setTranslateY = this.sinon.stub();
+    currentY = 0;
+    motion = new SwipeToKillMotion(el, {
+      setTranslateY(newY) {
+        currentY = newY;
+        setTranslateY(currentY);
+      }
     });
+  });
 
-    test('no icon available from appWindow', function() {
-      var card = this.card;
-      var app = card.app;
-      card.render();
-      getIconPromise.mFulfillToValue();
-
-      assert.ok(app.getSiteIconUrl.called);
-      assert.isFalse(card.iconButton.classList.contains('pending'));
-      assert.equal(card.iconButton.style.backgroundImage, '');
-    });
+  teardown(function() {
 
   });
 
-  suite('destroy', function() {
-    setup(function(){
-      this.card = new Card({
-        app: makeApp({ name: 'dummyapp' }),
-        manager: mockManager,
-        containerElement: mockManager.cardsList
+  function touch(subtype, x, y) {
+    var evt = new CustomEvent('touch' + subtype);
+    evt.touches = evt.targetTouches = [];
+    if (arguments.length > 1) {
+      evt.touches.push({
+        clientX: x, clientY: y
       });
-      this.card.render();
-    });
-
-    test('removes element from parentNode', function() {
-      var cardNode = this.card.element;
-      assert.ok(cardNode.parentNode, 'cardNode has parentNode when rendered');
-      this.card.destroy();
-      assert.ok(!cardNode.parentNode, 'cardNode has no parentNode');
-      assert.equal(cardsList.childNodes.length, 0, 'cardsList has no children');
-    });
+    }
+    return evt;
+  }
 
-    test('cleans up references', function() {
-      this.card.destroy();
-      assert.ok(!this.card.manager, 'card.manager reference is falsey');
-      assert.ok(!this.card.app, 'card.app reference is falsey');
-      assert.ok(!this.card.element, 'card.element reference is falsey');
-    });
+  test('swipe up to kill', function() {
+    var dropStub = this.sinon.stub();
```

Comment:
if (and only if) it's as simple as a search and replace I'd like to use real dispatch events here too